Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -679,7 +681,9 @@ namespace AzToolsFramework
             {
                 AZ::TransformBus::Event(entityId, &AZ::TransformInterface::SetWorldTM, transform);
             }
-            
+
+            // Update the undo cache on the added entity with the parent information.
+            m_prefabUndoCache.UpdateCache(entityId);
```

Comment:
I noticed that performing the update only on the added entity is enough because in `UpdateCache()` it can query the parent id and generate the DOM.

However, I prefab adding `m_prefabUndoCache.UpdateCache(parentId)` following this line to update the parent cache. Without this update, `EditorEntitySortComponent` is empty. It shouldn't be.